app/controllers/candidate_interface/sign_up_controller.rb

```diff
@@ -29,7 +32,12 @@ def show; end
   private
 
     def candidate_sign_up_form_params
-      params.require(:candidate_interface_sign_up_form).permit(:email_address, :accept_ts_and_cs)
+      params.require(:candidate_interface_sign_up_form).permit(:email_address, :accept_ts_and_cs).merge(course_id: @course_id)
+    end
+
+    def build_course
```

Comment:
Maybe this could be named `course_id` and be used in `candidate_sign_up_form_params`?

```ruby
def course_id
  @provider = Provider.find_by(code: params[:providerCode])
  @course = @provider.courses.find_by(code: params[:courseCode])
  if @course.present?
    @course.id
  else
    nil
  end
end
```

(Rubocop will probably complain about this code)

Reply:
Good suggestion. Have actioned 👍